sql/core/src/main/java/org/apache/spark/sql/vectorized/ColumnVector.java

```diff
@@ -215,10 +215,18 @@ public final ColumnarRow getStruct(int rowId) {
 
   /**
    * Returns the map type value for rowId.
+   *
+   * In Spark, map type value is basically a key data array and a value data array. A key from the
+   * key array with a index and a value from the value array with the same index contribute to
+   * an entry of this map type value.
+   *
+   * To support map type, implementations must construct an {@link ColumnarMap} and return it in
+   * this method. {@link ColumnarMap} requires a {@link ColumnVector} that stores the data of all
+   * the keys of all the maps in this vector, and another {@link ColumnVector} that stores the data
+   * of all the values of all the maps in this vector, and an offset and length which specifies the
```

Comment:
nit: ` an offset and length` -> `a pair of offset and length`? Or `specifies` -> `specify`?